lib/semantics/check-omp-structure.h

```diff
@@ -131,8 +133,11 @@ class OmpStructureChecker : public virtual BaseChecker {
     std::multimap<OmpClause, const parser::OmpClause *> clauseInfo;
   };
   // back() is the top of the stack
-  const OmpContext &GetContext() const { return ompContext_.back(); }
-  const OmpContext &GetPrevContext() const { return ompContext_.rbegin()[1]; }
+  // TODO: remove CHECK after all directives/clauses are checked
+  const OmpContext &GetContext() const {
+    CHECK(!ompContext_.empty());
+    return ompContext_.back();
+  }
   void SetContextDirectiveSource(const parser::CharBlock &directive) {
     ompContext_.back().directiveSource = directive;
```

Comment:
Use `GetContext()` here instead of `ompContext_.back()` so that it is checked here too.

Reply:
But here `GetContext` is returning a constant reference... Probably better to add `CHECK` for all the setters?

Reply:
It would be better to have a non-const overload of GetContext(), like this:
```
  OmpContext &GetContext() {
    CHECK(!ompContext_.empty());
    return ompContext_.back();
  }
  const OmpContext &GetContext() const {
    return const_cast<OmpStructureChecker *>(this)->GetContext();
  }
```

Or, if you never call GetContext from a const member function, you could omit the second one.